### PR TITLE
chore: use newest node-gyp for SDL pipeline

### LINKED ISSUE #195241: Improve custom node-gyp build step
This is so weird: https://github.com/microsoft/vscode/blob/31a16529b6a0cfb0ddb5be553407efa9042dc73f/build/azure-pipelines/win32/product-build-win32.yml#L96-L110

If we need to fix somethign in node-gyp, we should do what we always do: let's fork it, patch it, call it `@vscode/node-gyp`, publish it to npm and consume it as a dev dependency instead of this convoluted approach.

cc @sbatten

### PR DESCRIPTION
**Pull request by rzhao271**
This PR changes the SDL pipeline from using a custom fork of node-gyp to using the latest upstream node-gyp, now that the BinSkim fix has made it into the upstream package.

Fixes #195241

### CODE REVIEW

**Comment by deepak1556** [comment]
Can you also make the same change for https://github.com/microsoft/vscode/blob/8c24af1bf467eba4725a6b058f37e2faa43a6d15/build/azure-pipelines/win32/product-build-win32.yml#L92-L117

**Comment by rzhao271** [comment]
Confirmation builds
- SDL https://dev.azure.com/monacotools/Monaco/_build/results?buildId=252495&view=results
- Product build https://dev.azure.com/monacotools/Monaco/_build/results?buildId=252496&view=results

File: build/azure-pipelines/win32/product-build-win32.yml [hunk 1]
```diff
@@ -89,33 +89,17 @@ steps:
...
       $ErrorActionPreference = "Stop"
-      $env:npm_config_node_gyp="$(Join-Path $pwd.Path '.build/node-gyp/bin/node-gyp.js')"
-      $env:npm_config_arch="$(VSCODE_ARCH)"
+      # TODO: remove custom node-gyp when updating to Node v20,
+      # refs https://github.com/npm/cli/releases/tag/v10.2.3 which is available with Node >= 20.10.0
+      $nodeGypDir = "$(Agent.TempDirectory)/custom-packages"
+      mkdir "$nodeGypDir"
+      npm install node-gyp@10.0.1 -g --prefix "$nodeGypDir"
+      $env:npm_config_node_gyp = "${nodeGypDir}/node_modules/node-gyp/bin/node-gyp.js"
+      $env:npm_config_arch = "$(NPM_ARCH)"
```
Comment (deepak1556) [inline build/azure-pipelines/win32/product-build-win32.yml:102]: `NPM_ARCH` is not defined parameter for windows pipeline, continue to use `VSCODE_ARCH`

**Review by deepak1556 — CHANGES_REQUESTED** [review]
Since there are no `yarn.lock` changes install dependencies step will pull from cache. Please update the value in `build/.cachesalt` to force a rebuild of modules in validating the changes from this PR.

**Comment by rzhao271** [comment]
Moving to February debt week.

New verification builds:
SDL build https://dev.azure.com/monacotools/Monaco/_build/results?buildId=252763&view=results
Product build https://dev.azure.com/monacotools/Monaco/_build/results?buildId=252902&view=results

**Comment by deepak1556** [comment]
@rzhao271 you can rebase this PR on top of latest `main` to address the failing linux builds issues which got addressed in https://github.com/microsoft/vscode/commit/1396f9c8274692e5ed6f7089a62297f89238f894

**Comment by rzhao271** [comment]
New validation builds

VS Code SDL Build https://dev.azure.com/monacotools/Monaco/_build/results?buildId=254009
VS Code Product Build https://dev.azure.com/monacotools/Monaco/_build/results?buildId=254010

**Review by deepak1556 — APPROVED** [review]

**Pull request merged**